CLaSh tries to synthesize the string in errorX

With the latest version of CLaSH (b8440afa1289593772c60e49f04ec5a000b59bc3) it seems the string passed to `errorX` is not discarded before synthesis. This can cause problems if the string expression is not a literal. For example:

```
module Top where

import Clash.Prelude

{-# NOINLINE topEntity #-}
{-# ANN topEntity
  (Synthesize
    { t_name   = "Bug"
    , t_inputs =
          [ PortName "CLK_25MHZ"
          , PortName "RESET"
          , PortName "INPUT"
          ]
    , t_output = PortName "OUTPUT"
    }) #-}
topEntity
    :: Clock System Source
    -> Reset System Asynchronous
    -> Signal System Bit
    -> Signal System Bit
topEntity = exposeClockReset $ fmap $ \b -> errorX ("foo" <> show b)

```

The error message is:

```
Top.hs:21:1: error:
    Clash.Netlist.BlackBox(225): No blackbox found for: GHC.CString.unpackAppendCString#. Did you forget to include directories containing primitives? You can use '-i/my/prim/dir' to achieve this.
    
    NB: The source location of the error is not exact, only indicative, as it is acquired after optimisations.
    The actual location of the error can be in a function that is inlined.
    To prevent inlining of those functions, annotate them with a NOINLINE pragma.
   |
21 | topEntity = exposeClockReset $ fmap $ \b -> errorX ("foo" <> show b)
   | ^^^^^^^^^
```

--- a/clash-lib/src/Clash/Normalize/Transformations.hs
+++ b/clash-lib/src/Clash/Normalize/Transformations.hs
@@ -760,7 +760,7 @@ removeUnusedExpr _ e@(collectArgs -> (p@(Prim nm pty),args)) = do
          else changed (mkApps p args')
     _ -> return e
   where
-    arity = length . Either.lefts . fst $ splitFunForallTy pty
+    arity = length . Either.rights . fst $ splitFunForallTy pty
 
     go _ _ _ [] = return []
     go tcm n used (Right ty:args') = do

